js_modules/dagster-ui/packages/ui-core/src/partitions/__tests__/OrdinalOrSingleRangePartitionSelector.test.tsx

```diff
@@ -0,0 +1,174 @@
+import {PartitionHealthQuery} from '../../assets/types/usePartitionHealthData.types';
+import {
+  PartitionDimensionSelection,
+  buildPartitionHealthData,
+} from '../../assets/usePartitionHealthData';
+import {mockViewportClientRect, restoreViewportClientRect} from '../../testing/mocking';
+import {OrdinalOrSingleRangePartitionSelector} from '../OrdinalOrSingleRangePartitionSelector';
+
+const Wrapper = ({
+  queryResult,
+  initialSelection,
+}: {
+  queryResult: PartitionHealthQuery;
+  initialSelection?: PartitionDimensionSelection | null;
+}) => {
+  const assetHealth = buildPartitionHealthData(queryResult, {path: ['asset']});
+  const [selection, setSelection] = React.useState<PartitionDimensionSelection | null | undefined>(
+    initialSelection,
+  );
+  return (
+    <>
+      <OrdinalOrSingleRangePartitionSelector
+        dimension={assetHealth.dimensions[0]!}
+        health={assetHealth}
+        selection={selection}
+        setSelection={setSelection}
+      />
+      <div data-testid="selection">
+        {selection
+          ? JSON.stringify({
+              selectedKeys: selection.selectedKeys,
+              selectedRanges: selection.selectedRanges,
+            })
+          : selection === null
+          ? 'null'
+          : selection === undefined
+          ? 'undefined'
+          : ''}
+      </div>
+    </>
+  );
+};
+describe('OrdinalOrSingleRangePartitionSelector', () => {
+  beforeAll(() => {
+    mockViewportClientRect();
+  });
+  afterAll(() => {
+    restoreViewportClientRect();
+  });
+
+  it('should populate with an existing value (selectedKeys)', async () => {
+    const {getByText, getByTitle} = render(
+      <Wrapper
+        queryResult={TWO_DIMENSIONAL_ASSET_BOTH_STATIC}
+        initialSelection={{dimension: {} as any, selectedKeys: ['CA', 'MN'], selectedRanges: []}}
+      />,
+    );
+    expect(getByText('Single').parentElement).toHaveAttribute('aria-selected', 'true');
+    expect(getByTitle('CA')).toBeVisible(); // tags are in the element
+    expect(getByTitle('MN')).toBeVisible();
+  });
+
+  it('should populate with an existing value (null)', async () => {
+    const {getByText} = render(
+      <Wrapper queryResult={TWO_DIMENSIONAL_ASSET_BOTH_STATIC} initialSelection={null} />,
+    );
+    expect(getByText('All').parentElement).toHaveAttribute('aria-selected', 'true');
+    expect(getByText('5 partitions')).toBeVisible();
+  });
+
+  it('should support All or Single entry for static partition dimensions', async () => {
+    const {getByText, getByTestId, queryByText} = render(
+      <Wrapper queryResult={TWO_DIMENSIONAL_ASSET_BOTH_STATIC} />,
+    );
+    expect(getByText('All')).toBeVisible();
+    expect(getByText('Single')).toBeVisible();
+    expect(queryByText('Range')).not.toBeInTheDocument();
+
+    expect(getByTestId('selection').textContent).toEqual('undefined');
+
+    // Click All, verify the value changes to `null`
+    await userEvent.click(getByText('All'));
```

Comment:
The new recommended pattern is `const user = userEvent.setup()` and then `await user.click(...)`. This results in better isolated tests.

Reply:
Ahh interesting will update this!